src/fraction.ts

```diff
@@ -134,127 +132,111 @@ export class Fraction {
+    let otherDenominator: number;
 
     if (param1 instanceof Fraction) {
       otherNumerator = param1.numerator;
       otherDenominator = param1.denominator;
     } else {
-      if (param1 !== undefined) {
-        otherNumerator = param1;
-      } else {
-        otherNumerator = 1;
-      }
-
-      if (param2 !== undefined) {
-        otherDenominator = param2;
-      } else {
-        otherDenominator = 1;
-      }
+      otherNumerator = param1;
+      otherDenominator = param2;
     }
 
     return this.set(this.numerator * otherDenominator, this.denominator * otherNumerator);
   }
 
-  // Simplifies both sides and checks if they are equal.
-  equals(compare) {
+  /** Simplifies both sides and checks if they are equal. */
+  equals(compare: Fraction | number): boolean {
     const a = Fraction.__compareA.copy(compare).simplify();
     const b = Fraction.__compareB.copy(this).simplify();
 
     return a.numerator === b.numerator && a.denominator === b.denominator;
   }
 
-  // Greater than operator.
-  greaterThan(compare) {
+  /** Greater than operator. */
+  greaterThan(compare: Fraction | number): boolean {
     const a = Fraction.__compareB.copy(this);
     a.subtract(compare);
     return a.numerator > 0;
   }
 
-  // Greater than or equals operator.
-  greaterThanEquals(compare) {
+  /** Greater than or equals operator. */
+  greaterThanEquals(compare: Fraction | number): boolean {
     const a = Fraction.__compareB.copy(this);
     a.subtract(compare);
     return a.numerator >= 0;
   }
 
-  // Less than operator.
-  lessThan(compare) {
+  /** Less than operator. */
+  lessThan(compare: Fraction | number): boolean {
     return !this.greaterThanEquals(compare);
   }
 
-  // Less than or equals operator.
-  lessThanEquals(compare) {
+  /** Less than or equals operator. */
+  lessThanEquals(compare: Fraction | number): boolean {
     return !this.greaterThan(compare);
   }
 
-  // Creates a new copy with this current values.
-  clone() {
+  /** Creates a new copy with this current values. */
+  clone(): Fraction {
     return new Fraction(this.numerator, this.denominator);
   }
 
-  // Copies value of another Fraction into itself.
-  copy(copy) {
+  //** Copies value of another Fraction into itself. */
+  copy(copy: Fraction | number): this {
     if (typeof copy === 'number') {
       return this.set(copy || 0, 1);
     }
     return this.set(copy.numerator, copy.denominator);
   }
 
-  // Returns the integer component eg. (4/2) == 2
-  quotient() {
+  /** Returns the integer component eg. (5/2) == 2 */
+  quotient(): number {
     return Math.floor(this.numerator / this.denominator);
   }
 
-  // Returns the fraction component when reduced to a mixed number
-  fraction() {
+  //** Returns the fraction component when reduced to a mixed number */
+  fraction(): number {
     return this.numerator % this.denominator;
   }
 
-  // Returns the absolute value
-  abs() {
+  //** Returns the absolute value */
```

Comment:
This comment is misleading. It does not only return the absolute value, it modifies the object as well.
What is right the comment or the implementation?

Reply:
This comment has raised the issue. I suggest to add comments to the functions without them. @0xfe @mscuthbert I will start with the obvious ones but I will need your help on those for `tuning`.

Reply:
@0xfe in reality what was wrong is the comment, not the implementation. I will leave a note in the code to discuss about it. This abs function is only used in this file.

Reply:
Right, it's just unexpected behavior -- I'd suggest renaming to `makeAbs` and updating references.